lstchain/io/io.py

```diff
@@ -298,8 +298,11 @@ def auto_merge_h5files(
         source_filenames.append(str(file))
 
     with open_file(output_filename, mode="a") as file:
-        sources = file.create_group("/", "source_filenames", "List of files merged")
-        file.create_array(sources, "filenames", source_filenames, "List of files merged")
+        try:
+            file.root.source_filenames._f_remove(recursive=True)
+        except tables.NoSuchNodeError:
+            sources_group = file.create_group("/", "source_filenames", "List of files merged")
```

Comment:
This now only creates the source filenames if it was already there.

I think you should switch to:

```
if "/source_filenames" in f.root:
    f.remove_node("/", "source_filenames", recursive=True)

sources_group = file.create_group("/", "source_filenames", "List of files merged")
 file.create_array(sources_group, "filenames", source_filenames, "List of files merged")
```
```

Reply:
@maxnoe 
I guess you mean that data is written only if the node `root.source_filenames` **does not** exist in the file, so no data is written if `root.source_filenames` is already there. You're right, I've fixed the code.